cmd/wait-for-github/pr_test.go

```diff
@@ -90,12 +91,30 @@ func TestPRCheck(t *testing.T) {
 			},
 		},
 		{
-			name: "Not merged, but CI failed",
+			name: "Not merged, but CI failed (stop-on-failed-ci enabled)",
 			fakeClient: fakeGithubClientPRCheck{
 				CIStatus: github.CIStatusFailed,
 			},
 			expectedExitCode: &one,
 		},
+		{
+			name: "Merged, CI failed",
+			fakeClient: fakeGithubClientPRCheck{
+				CIStatus:     github.CIStatusFailed,
+				MergedCommit: "abc123",
+				MergedAt:     1234567890,
+			},
+			expectedExitCode: &zero,
+		},
+		{
+			name: "Closed, CI failed (stop-on-failed-ci disabled)",
+			fakeClient: fakeGithubClientPRCheck{
+				Closed:   true,
+				CIStatus: github.CIStatusFailed,
+			},
+			allowFailedCI:    true,
```

Comment:
Instead of using a test-only field here, I think it makes more sense to directly set the `prConfig` field for the test scenario:

```suggestion
			continueOnFailedCI:    false,
```

Reply:
c4389f1